src/v2/utils/dataGridUtils.js

```diff
@@ -167,3 +168,18 @@ export const formatGridColumnsForConfig = (gridColumns, defaultCheckedValue) =>
 	});
 	return formattedGridColumns;
 };
+
+export const setSelectedRowsAsUnchecked = ({ checkIds, studentList }) => {
+	return studentList?.map((student) => {
+		let studentObject = { ...student };
+		checkIds.forEach((checkedId) => {
```

Comment:
Avoid using nested iterations, This will slow down the app.
```suggestion
		checkedIds.includes(student.id)
```